config/config.php

```diff
@@ -1,33 +1,33 @@
 <?php
 
-use Zend\Stdlib\ArrayUtils;
-use Zend\Stdlib\Glob;
+use Zend\ConfigAggregator\ConfigAggregator;
+use Zend\ConfigAggregator\PhpFileProvider;
 
-/**
- * Configuration files are loaded in a specific order. First ``global.php``, then ``*.global.php``.
- * then ``local.php`` and finally ``*.local.php``. This way local settings overwrite global settings.
- *
- * The configuration can be cached. This can be done by setting ``config_cache_enabled`` to ``true``.
- *
- * Obviously, if you use closures in your config you can't cache it.
- */
+// To enable or disable caching, set the `ConfigAggregator::ENABLE_CACHE` boolean in
+// `config/autoload/local.php`.
+$cacheConfig = [
+    'config_cache_path' => 'data/config-cache.php',
+];
 
-$cachedConfigFile = 'data/cache/app_config.php';
+$aggregator = new ConfigAggregator([
+    //Include cache config for zf-development-mode
+    function () use ($cacheConfig) {
+        return $cacheConfig;
+    },
```

Comment:
I've created zendframework/zend-config-aggregator#1 to track this. I think we can solve this using the current architecture via a new `ArrayProvider` class:

```php
$configCache = ['config_cache_path' => 'data/config-cache.php'];

$aggregator = new ConfigAggregator([
    new \Zend\ConfigAggregator\ArrayProvider($configCache),
    /* ... */
], $configCache['config_cache_path']);
```

It's a little extra overhead, but keeps the current semantics, and prevents code complexity from growing.